File: Models/PluginManifest.cs
using System;

namespace ClassIsScore.Models;

/// <summary>
/// 插件元数据模型
/// </summary>
public class PluginManifest
{
    public string Id { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string Version { get; set; } = string.Empty;
    public string Author { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string EntranceAssembly { get; set; } = string.Empty;
}

File: server/src/main/java/com/classisscore/server/plugin/PluginManifest.java
package com.classisscore.server.plugin;

import lombok.Data;
import lombok.Builder;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class PluginManifest {

    private Long id;
    private String name;
    private String version;
    private String description;
    private String author;
    private String mainClass;
    private Boolean enabled;
    private String filePath;
}
